bin/swap_diversity_keys.sh

```diff
@@ -0,0 +1,111 @@
+            exit
+        fi
+    fi
+fi
+
+if [ ! -f "$PRIVATE_KEY_FILE" ]; then
+    echoerr "Private key file does not exist: $PRIVATE_KEY_FILE"
+    exit
+elif ! grep -q 'BEGIN PGP PRIVATE KEY BLOCK' $PRIVATE_KEY_FILE; then
+    echoerr "$PRIVATE_KEY_FILE does not contain a private key"
+    exit
+elif [ ! -f $PUBLIC_KEY_FILE ]; then
+    echoerr "Public key file does not exist: $PUBLIC_KEY_FILE"
+    exit
+elif ! grep -q 'BEGIN PGP PUBLIC KEY BLOCK' $PUBLIC_KEY_FILE; then
+    echoerr "$PUBLIC_KEY_FILE does not contain a public key"
+    exit
+fi
+
+echo "------------------------------------------------------------------------------------------"
+echo "NAMESPACE: $NAMESPACE"
+echo "SECRET NAME: $SECRET_NAME"
+echo "BACKUP SECRET NAME: $BACKUP_SECRET_NAME"
+echo "PRIVATE KEY FILE: $PRIVATE_KEY_FILE"
+echo "PUBLIC KEY FILE: $PUBLIC_KEY_FILE"
+read -p "ARE YOU SURE YOU WANT TO CONTINUE(YES/NO)" CONFIRM
+if [ "$CONFIRM" != "YES" ]; then
+    echoerr "BYE"
+    exit
+fi
+echo "------------------------------------------------------------------------------------------"
+
+kubens $NAMESPACE
+BACKUP_SECRET_EXISTS=$(kubectl get --ignore-not-found secret $BACKUP_SECRET_NAME)
+if [ "$BACKUP_SECRET_EXISTS" != "" ]; then
+     echoerr "There is already a secret with the name $BACKUP_SECRET_NAME"
+    exit
+fi
+CURRENT_KEY_FILENAME="diversity-current-key-$(date '+%Y-%m-%dT%H:%M:%S').key"
+if [ -f $CURRENT_KEY_FILENAME ]; then
+    echoerr "backup file already exists $CURRENT_KEY_FILENAME"
+    exit
+fi
+CURRENT_SECRET_CONTENT=$(kubectl get secret $SECRET_NAME -o json)
+if [ `command -v jq` ]; then
+  CURRENT_SECRET_CONTENT=$(echo $CURRENT_SECRET_CONTENT | jq '.data|map_values(@base64d)' --raw-output)
+fi
+echo $CURRENT_SECRET_CONTENT > $CURRENT_KEY_FILENAME
+echo "Saved contents of secret $SECRET_NAME to $CURRENT_KEY_FILENAME"
+
+echo "Copying $SECRET_NAME secret to $BACKUP_SECRET_NAME secret"
+KS_COMMAND="kubectl get secret $SECRET_NAME -o yaml |  sed -e '/uid:.*/d' -e '/resourceVersion: .*/d' -e '/creationTimestamp: .*/d' -e 's/name: $SECRET_NAME/name: $BACKUP_SECRET_NAME/g' | kubectl apply -f -"
+echo "Running $KS_COMMAND"
+eval $KS_COMMAND
+echo "Dumping contents of backup secret $BACKUP_SECRET_NAME"
+kubectl get secret $BACKUP_SECRET_NAME -o yaml
+
+echo "------------------------------------------------------------------------------------------"
+echo "Putting site into maintenance mode"
+# Sometimes the vpn makes the maintenance mode script fail even if it has successfully put the site into maintenance mode
+./bin/maintenance_mode.sh $ENVIRONMENT True || true
+
+echo "Removing the old keys by deleting the secret $SECRET_NAME"
+kubectl delete secret $SECRET_NAME
+echo "Recreating $SECRET_NAME with new keys"
+kubectl create secret generic $SECRET_NAME --from-file=private=$PRIVATE_KEY_FILE --from-file=public=$PUBLIC_KEY_FILE
+echo "Restarting deployment so pods pickup new secret"
+kubectl rollout restart deployment cla-backend-app
```

Comment:
Is it worth mentioning somewhere (even in the PR text) that this rolling restart won't work for a branch deployment in uat?

Reply:
updated description to read: Puts the site into maintenance mode (this doesn't work on branch deployments)